src/mkpsxiso/miniaudio_pcm.h

```diff
@@ -0,0 +1,225 @@
+
+static size_t virtual_wav_read(ma_decoder *pDecoder, void *pBufferOut, size_t bytesToRead)
+{
+    VirtualWav *vw = (VirtualWav *)pDecoder->pUserData;
+    size_t bytesRead = 0;
+    if(vw->vpos < 44)
+    {
+        const size_t headerread = drwav_min(bytesToRead, 44-vw->vpos);
+        memcpy(pBufferOut, &vw->header[vw->vpos], headerread);
+        vw->vpos += headerread;
+        bytesRead += headerread;
+        bytesToRead -= headerread;
+        pBufferOut = ((uint8_t*)pBufferOut) + headerread;
+    }
+    if(bytesToRead > 0)
+    {
+        const size_t actualread = fread(pBufferOut, 1, bytesToRead, vw->file);
+        bytesRead += actualread;
+        vw->vpos += actualread;
+        vw->pos += actualread;
+    }
+    return bytesRead;
+}
+
+static ma_bool32 virtual_wav_seek(ma_decoder *pDecoder, ma_int64 byteOffset, ma_seek_origin origin)
+{
+    int whence;
+    int result;
+    VirtualWav *vw = (VirtualWav *)pDecoder->pUserData;
+
+    if (origin == ma_seek_origin_start) {
+        if(byteOffset < 0) return MA_FALSE;
+        if(byteOffset > vw->vsize) return MA_FALSE;
+        vw->vpos = byteOffset;
+        byteOffset = drwav_max(byteOffset - 44, 0);
+        vw->pos = byteOffset;
+        whence = SEEK_SET;
+    } else if (origin == ma_seek_origin_end) {
+        if(byteOffset > 0) return MA_FALSE;
+        if((byteOffset + vw->vsize) < 0) return MA_FALSE;
+        vw->vpos = vw->vsize + byteOffset;
+        byteOffset = drwav_max(byteOffset, -(vw->vsize - 44));
+        vw->pos = (vw->vsize - 44) + byteOffset;
+        whence = SEEK_END;
+    } else {
+        if((byteOffset+vw->vpos) > vw->vsize) return MA_FALSE;
+        if((byteOffset+vw->vpos) < 0) return MA_FALSE;
+        vw->vpos += byteOffset;
+        uint64_t abspos = vw->pos + byteOffset;
+        if(abspos < 0)
+        {
+            byteOffset = -vw->pos;
+        }
+        else if(abspos > (vw->vsize-44))
+        {
+            byteOffset = (vw->vsize-44) - vw->pos;
+        }
+        vw->pos += byteOffset;
+        whence = SEEK_CUR;
+    }
+
+#if defined(_WIN32)
+    #if defined(_MSC_VER) && _MSC_VER > 1200
+        result = _fseeki64(vw->file, byteOffset, whence);
+    #else
+        /* No _fseeki64() so restrict to 31 bits. */
+        if (origin > 0x7FFFFFFF) {
+            return MA_FALSE;
+        }
+
+        result = fseek(vw->file, (int)byteOffset, whence);
+    #endif
+#else
+    result = fseek(vw->file, (long int)byteOffset, whence);
+#endif
+    if (result != 0) {
+        return MA_FALSE;
+    }
+
+    return MA_TRUE;
+}
+
+#if !defined(_MSC_VER) && !((defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 1) || defined(_XOPEN_SOURCE) || defined(_POSIX_SOURCE)) && !defined(MA_BSD)
+int fileno(FILE *stream);
+#endif
+
+static ma_result stdio_file_size(FILE *file, uint64_t *pSizeInBytes)
+{
+    int fd;
+    struct stat info;
+
+    MA_ASSERT(file  != NULL);
+
+#if defined(_MSC_VER)
+    fd = _fileno((FILE*)file);
+#else
+    fd =  fileno((FILE*)file);
```

Comment:
Are those typecasts needed? `file` is already of type `FILE*`.

Reply:
No, leftover from copy paste, forgot to remove.